### ISSUE TITLE
[BUG] RIN and Likelihood models

### ISSUE THREAD

**Issue by PaulRabich** [unit8co/darts#2743]
**Describe the bug**
When using a Torch-based model with `use_reversible_instance_norm=True` and a `likelihood` specified (e.g., GaussianLikelihood), the model's predicted distribution parameters appear to be denormalized by the Reversible Instance Normalization (RINorm). This seems unintended, as RINorm should generally only affect the forecasted target values, not learned distribution parameters.

**To Reproduce**
```python
from darts.models import  NLinearModel
from darts.utils.likelihood_models import GaussianLikelihood

# Constant TS creation
constant_two = np.ones(100) * 2
constant_two_series = TimeSeries.from_values(constant_two)

# Create normal model without likelihood and with reversible instance norm
model_normal = NLinearModel(20, 10, use_reversible_instance_norm=False)
model.fit(constant_two_series)
normal_pred_series = model.predict(20, series=constant_two_series)


# Create model with likelihood and with reversible instance norm
model_likelihood_with_revin = NLinearModel(20, 10, use_reversible_instance_norm=True, likelihood=GaussianLikelihood())
model_likelihood_with_revin.fit(constant_two_series)
likelihood_with_revin_pred_series = model_likelihood_with_revin.predict(20, series=constant_two_series)

# Create model with likelihood and with reversible instance norm
model_likelihood = NLinearModel(20, 10, use_reversible_instance_norm=False, likelihood=GaussianLikelihood())
model_likelihood.fit(constant_two_series)
likelihood_pred_series = model_likelihood.predict(20, series=constant_two_series)

# Create side-by-side plots
fig, axes = plt.subplots(1, 3, figsize=(14, 5)) 

# Left plot: Normal model
constant_two_series.plot(ax=axes[0], label="constant")
normal_pred_series.plot(ax=axes[0], label="normal")
axes[0].set_title("NLinear (no likelihood, with RevIN)")
axes[0].legend()

# Right plot: Likelihood model
constant_two_series.plot(ax=axes[1], label="constant")
likelihood_with_revin_pred_series.plot(ax=axes[1], label="likelihood")
axes[1].set_title("NLinear (with Gaussian Likelihood, with RevIN)")
axes[1].legend()

# Right plot: Likelihood model
constant_two_series.plot(ax=axes[2], label="constant")
likelihood_pred_series.plot(ax=axes[2], label="likelihood")
axes[2].set_title("NLinear (with Gaussian Likelihood, without RevIN)")
axes[2].legend()


plt.tight_layout()
plt.show()
```

![Image](https://github.com/user-attachments/assets/fb569a9d-9c17-4cc0-a783-78dfc98b4400)

**Expected behavior**
While denormalizing the predicted output values makes sense in standard forecasting scenarios, applying this 
enormalization to distribution parameters (like the mean or variance from a likelihood model) is typically not appropriate. 
These parameters should remain in the normalized space unless explicitly required otherwise.

**System (please complete the following information):**
 - Python version: 3.11.11
 - darts version: 0.34.0

**Additional context**
When inspecting the code responsible for applying RINorm—specifically in `darts.models.forecasting.pl_forecasting_module`, lines 48 to 67—it appears that all outputs of the model are denormalized. 
[Link to Code](https://github.com/unit8co/darts/blob/42776790183bbe42411fc2dba3e1e8416f9263e8/darts/models/forecasting/pl_forecasting_module.py#L48C5-L67C27)

**Comment by dennisbader** [comment]
Hi @PaulRabich. Yes, currently reversible instance norm should not be used with likelihoods other than `QuantileRegression` (and maybe `PoissonDistribution` or others that return the mean only).

Apart from that, your using a probabilistic model (since you use a `likelihood`), meaning that at prediction time, you should either generate sample predictions with `num_samples>>1` or predict the likelihood/distribution parameters directly with `predict_likelihood_parameters=True, num_samples=1`.

By default, `predict()` uses `num_samples=1` which in case of probabilistic models just draws one (random) sample from the distribution. This is why you get such bumpy forecasts.
